lib/fathead/django_built_ins/parse.py

```diff
@@ -0,0 +1,177 @@
+
+    def parse_code_from_data(self, section):
+        """
+        Look for an example code block to output
+        Args:
+            section: A section of parsed HTML that represents a templatetag or filter
+
+        Returns:
+            Formatted code string
+        """
+        code = section.find('div', {'class': 'highlight'})
+        if code:
+            return '{}{}{}'.format('<pre><code>', code.text, '</code></pre>')
+        return ''
+
+    def parse_for_data(self):
+        """
+        Main gateway into parsing the data. Will retrieve all necessary data elements.
+        """
+        data = []
+
+        for section in (self.tag_sections + self.filter_sections):
+            name, anchor = self.parse_name_and_anchor_from_data(section)
+            first_paragraph = self.parse_first_paragraph_from_data(section)
+            code = self.parse_code_from_data(section)
+
+            data_elements = {
+                'name': name,
+                'anchor': anchor,
+                'first_paragraph': first_paragraph,
+                'code': code
+            }
+
+            data.append(data_elements)
+
+        self.parsed_data = data
+
+    def get_data(self):
+        """
+        Get the parsed data.
+        Returns:
+            self.parsed_data: Dict containing necessary data elements
+        """
+        return self.parsed_data
+
+
+class DjangoDataOutput(object):
+    """
+    Object responsible for outputting data into the output.txt file
+    """
+    def __init__(self, data):
+        self.data = data
+
+    def create_file(self):
+        """
+        Iterate through the data and create the needed output.txt file.
+
+        """
+        with open('output.txt', 'w+') as output_file:
+            for data_element in self.data:
+                if data_element.get('name'):
+                    name = data_element.get('name')
+                    abstract = '{}<br>{}'.format(data_element.get('code'), data_element.get('first_paragraph'))
+                    url = '{}{}'.format(DJANGO_DOC_URL, data_element.get('anchor'))
+                    list_of_data = [
+                        name,       # unique name will be the name of the templatetag/filter
+                        'A',        # type is article
+                        '',         # no redirect data
+                        '',         # ignore
+                        '',         # no categories
+                        '',         # ignore
+                        '',         # no related topics
+                        '',         # ignore
+                        DJANGO_HOME,# add an external link back to Django home
+                        '',         # no disambiguation
+                        '',         # images
+                        abstract,   # abstract
+                        url         # url to templatetag/filter doc
+                    ]
+                    output_file.write('\n{}'.format('\t'.join(list_of_data)))
+
+
```

Comment:
I have no strong feelings either way. I can't imagine another module importing this one, but extra protection never hurts.